src/lib/storage/chunk.cpp

```diff
@@ -43,7 +43,7 @@ void Chunk::append(const std::vector<AllTypeVariant>& values) {
   DebugAssert(is_mutable(), "Can't append to immutable Chunk");
 
   // Do this first to ensure that the first thing to exist in a row is the MVCC data.
-  if (has_mvcc_data()) get_scoped_mvcc_data_lock()->grow_by(1u, TransactionID{0});
+  if (has_mvcc_data()) get_scoped_mvcc_data_lock()->grow_by(1u, true);
```

Comment:
`MvccData::GrowMode::VisibleForAll/HiddenForAll`?

Reply:
I get [the original idea](https://softwareengineering.stackexchange.com/questions/323554/are-all-boolean-arguments-flag-arguments-and-thus-a-code-smell) that bools that change the control flow might be a code flow, but this hate for bool values is going way too far. That is a single, simple parameter that does not significantly change any control flow. How is that different from `1u`?

Reply:
grow_by(1) is intuitive ("add one element at the end.") - grow_by(1, false) could mean anything. I think it's similar for many places where I'd rather see a named bool.

libhyrise is 42k lines. A dozen named bools are imo not excessive. I am not being pedantic when I say I cannot read `grow_by(1u, true);` - I really cannot.  :(

Reply:
In the worst case - specific to this situation - one might  read "grow_by(1, false)" as "add one false at the end"

Reply:
Or bring back the CommitID parameter that was previously there? If not, I'd even split it into two methods, just to avoid yet another parameter. I'm not exaggerating when I'm saying that having to look up the name of those enums while writing code is one of the top things that annoy me the most.

Reply:
another method would be fine. 
(Let's blame c++ for not having named parameters)

Reply:
That doesn't work for the constructor. :/
I'll revert to the CommitID that was there previously.

Reply:
> I'm not exaggerating when I'm saying that having to look up the name of those enums while writing code is one of the top things that annoy me the most.

I understand that will be the case if you know the function well, maybe even implemented it. I cannot imagine it would be like that if someone else wrote the function - or if someone else used a function you wrote.